conversion_sr14_sr15/conversion_sr14_sr15.py

```diff
@@ -0,0 +1,149 @@
+import pandas as pd
+import yaml
+import os
+
+class ConfigLoader:
+    def __init__(self, config_path):
+        self.config_path = config_path
+        self.config = self.load_config()
+
+    def load_config(self):
+        os.chdir(os.path.dirname(os.path.abspath(__file__)))
+        with open(self.config_path, 'r') as yamlfile:
+            return yaml.load(yamlfile, Loader=yaml.FullLoader)
+
+class DataLoader:
+    def __init__(self, config):
+        self.config = config
+        self.hh_s14 = None
+        self.per_s14 = None
+        self.mgra_xwalk = None
+        self.landuse_s14 = None
+        self.landuse_abm3 = None
+        self.load_data()
+
+    def load_data(self):
+        os.chdir(self.config['input']['input_dir'])
+        self.hh_s14 = pd.read_csv(self.config['input']['filenames']['households'])
+        self.per_s14 = pd.read_csv(self.config['input']['filenames']['persons'])
+        self.mgra_xwalk = pd.read_csv(self.config['input']['filenames']['mgra_xwalk'])
+        self.landuse_s14 = pd.read_csv(self.config['input']['filenames']['land_use'])
+        self.landuse_abm3 = pd.read_csv(self.config['input']['filenames']['land_use_abm3'])
+
+class Converter:
+    def __init__(self, data_loader):
+        self.data_loader = data_loader
+        self.mgra_xwalk_dict = self.data_loader.mgra_xwalk.set_index('MGRA13')['MGRA15'].to_dict()
+
+    def convert_hh(self):
+        hh_s14 = self.data_loader.hh_s14
+        hh_s14['mgra'] = hh_s14['mgra'].replace(self.mgra_xwalk_dict)
+        hh_s15_converted = hh_s14.rename(columns={'hworkers': 'num_workers'})
+        return hh_s15_converted
+
+    def convert_per(self):
+        per_s14 = self.data_loader.per_s14
+        per_s14.rename(columns={'miltary': 'military'}, inplace=True)
+        per_s15_converted = per_s14.drop(columns=['indcen', 'weeks', 'hours', 'rac1p', 'hisp', 'version'])
```

Comment:
SR15 persons file contains the columns "indcen,weeks,hours,rac1p,hisp,version,naics2_original_code,soc2"

Reply:
I updated the script to keep all the columns from SR14 persons file in the SR15 persons file. The log file mentions that naics2_original_code and soc2 are missing but they should be in the converted persons file.

Reply:
> I updated the script to keep all the columns from SR14 persons file in the SR15 persons file. The log file mentions that naics2_original_code and soc2 are missing but they should be in the converted persons file.

I believe the columns "naics2_original_code" and "soc2" are new in SR15, not available in SR14. Could you add those in the converted persons file?

Reply:
These two columns have been added by drawing a random sample from an existing SR 15 synthetic person file. A SR 15 synthetic person file is now a required input file.